### DISCUSSION TITLE
A new Windows client in 2024. MMC4W

### DISCUSSION THREAD

**Discussion by drgerg** [MusicPlayerDaemon/MPD#1966] [Show and tell]
Hello.  With fear and trembling I present my project: MMC4W - Minimal MPD Client for Windows.
https://github.com/drgerg/mmc4w/tree/main

I use it on a daily basis now.  I added it to the Graphical Client list on the MPD site.
Just thought I'd share.

**Comment by drgerg** [comment]
Minimal MPD Client - v0.9.4

Check out the offline help under the Help menu. (updated for v0.9.4)

- Added more Playlist options:
    - Create a new empty saved playlist
    - Remove a saved playlist
    - Toggle Art/Add changes the Art button to a green Add button. 
    - Use it to add the currently playing song to any saved playlist 
    - To add specific songs, search and play the song using the normal means. 
    - When the song you want is playing, push the Add button and select the Playlist.

More playlist functions are coming.

**Comment by drgerg** [comment]
I keep thinking I've got it pretty well finished, but not yet I guess.  I added two more playlist functions so I could see the songs in the last loaded playlist.  If I select one of those, after a safety prompt, it gets deleted.
So I have:
- Create a new playlist
- Add current song to any playlist
- List songs in current playlist
- Remove a song from that playlist
- Delete any saved playlist.
I'm feeling pretty good about it.  ¯\\_(ツ)_/¯

**Comment by drgerg** [comment]
v0.9.6 may be the last update for a while.  I think I have all the functions I expect in a music player in this little app.

**Comment by martadinata666** [comment]
Oh interesting, A simple request, can you make this also stream? I know this may out of context of player control, something like set a stream url and fetch audio. Thanks

**Reply by drgerg** [reply]
I can add selecting of outputs.  The configuring of those outputs happens in the server.  Because I only use one output, this was way down low on my list of things to do, but I will be adding output selection.

**Reply by martadinata666** [reply]
In my or maybe a few other people let mpd play all day on the server and stream the output via HTTP, if this app can also stream the HTTP out it reduces the clunkiness to run another instance of VLC just to stream audio.

**Reply by drgerg** [reply]
Ah, yeah I see what you're talking about.  That isn't going to ever be part of this app.

**Reply by drgerg** [reply]
I think myMPD does what you're asking about:  https://jcorporation.github.io/myMPD/references/local-playback

**Reply by drgerg** [reply]
Ok.  My curiosity is piqued.  I'm going to be thinking about this . . .

**Comment by drgerg** [comment]
I dug around a bit.  I tried python-vlc but that is a real mess from my perspective.
I found that I can implement Python's webbrowser module, and it does work.  It uses your default browser and opens the stream in a new tab.  Volume control happens in the browser.  
If your intent is to start it, set it and let it run, that should be just fine.

I'm going to get this incorporated in the next few days.

**Comment by drgerg** [comment]
Alright.  Basic support for playing http streams is added in v0.9.7 available here.
https://github.com/drgerg/mmc4w

MMC4W is not playing anything.  It's just creating a convenience link between the MPD server and your default browser.  But it does work as intended, so maybe it's better than nothing, which is pretty much my attitude toward listening to music over http anyway.  

That is said with a smile, not sarcastically.  Enjoy! Have a great day!

**Comment by drgerg** [comment]
Should I keep announcing new versions here?  I'm not sure.  ¯\\_(ツ)_/¯

v0.9.8 is available at https://github.com/drgerg/mmc4w/releases/tag/v0.9.8

Minimal MPD Client - v0.9.8

Check out the offline help under the Help menu. (updated for v0.9.8)

Accomplished four things that I really wanted:

- MMC4W plays nicer with other clients on the same server. 
    It is still intentionally a non-polling client, so when changes occur, 
    it finds out about them a bit later. Not too long, though.
- The configuration file allows you to set the window sizes for all the windows.
    Search, Playlist songs, and art.
- MMC4W's volume buttons follow the server if there are changes there.
- Any server updates applied by other clients can be synced by pushing Play.
- If another client pauses playback, if you push MMC4W's pause, it picks that up.
    It will also pick it up when the playback timer runs down.  So it all depends
    on how impatient you are to see the interface update the status.
- Added text to the Help with a stab at explaining the contents of the mmc4w.ini file.
- Made the Help window larger based on display height.
- Added duplicate checking before adding songs to saved playlists.

If I should stop posting this stuff here, just let me know.

**Reply by MaxKellermann** [reply]
You can post here if you want, but you can also post a PR to the MPD website repository to post news. (Sorry if it takes a day or two until I merge it - I'm extremely scarce on time ATM)
